### ISSUE TITLE
Question about result : a difference between the results in the paper and my results

### ISSUE THREAD

**Issue by minsu1206** [facebookresearch/banmo#51]
Thank you for nice work.

I have a question about the results in the paper.
I just followed your instruction for data processing, training, rendering and evaluation, which is described at `scripts/README.md`

Desired results should be same as 
![image](https://user-images.githubusercontent.com/59243570/229417954-b62dd977-9fcb-41ba-8f67-8a437b48dcbd.png)

BUT my models shows poor performance results.
I will describe my results below.

AMA

1. training model by using T_swing and T_samba simultaneously --> evaluation 
    :: ave 11.4 chamfer distance about T_swing
    :: ave 10.7 chamfer distance about T_samba
2. download some files by cmd `wget https://www.dropbox.com/sh/n9eebife5uovg2m/AAA1BsADDzCIsTSUnJyCTRp7a -O tmp.zip` --> evaluation (=same as your instruction) 
    :: ave 9.1 chamfer distance about T_swing
3. train model by using only T_swing videos --> evaluation (same process about T_samba, too)
   :: ave 8.6 chamfer distance about T_swing
   :: ave 8.2 chamfer distance about T_samba
   
Synthetic (hand, eagle)
:: ave 6.4 chamfer distance about eagle
:: ave 5.3 chamfer distance about hands

I used 2xA100 for training, and same conda environment as yours.

As you see, My all results show poor performance.
Here, My question is How can I get same results described in the paper?

Best regards.

**Comment by gengshan-y** [comment]
Hi, the results of this repo should be aligned with the [camera-ready version](https://openaccess.thecvf.com/content/CVPR2022/papers/Yang_BANMo_Building_Animatable_3D_Neural_Models_From_Many_Casual_Videos_CVPR_2022_paper.pdf), where Tab. 1 got updated for all entries.

However, I wouldn't expect the performance to drop when training setup 1., compared to setup 3. Does it visually look worse?

**Comment by minsu1206** [comment]
Thanks you for your answer.
I just downloaded papers from arxiv, so I didn't know camera-ready version. I will refer to this paper. Then, result of AMA setup2 makes sense. 

AMA setup 1 look worse than setup 3. Specifically, at mesh visualization, details of leg and skirt  are more accurate in setup 3 than setup 1. 

I will attach reference images below. (meshes are visualized by MeshLab) 

T_samba1-mesh-00000.obj from setup1
![image](https://user-images.githubusercontent.com/59243570/229533910-a7d186b7-8646-4263-96b5-4d734647b825.png)


T_samba1-mesh-00000.obj from setup3
![image](https://user-images.githubusercontent.com/59243570/229534013-02ea3ae9-ae16-4cd7-9cea-0d2a0131a73b.png)


T_swing1-mesh-00000.obj from setup1
![image](https://user-images.githubusercontent.com/59243570/229539111-0ca9ef47-7a89-4591-9147-0bf3a851c63f.png)


T_swing1-mesh-00000.obj from setup3
![image](https://user-images.githubusercontent.com/59243570/229539271-fcc8f48e-87d3-450b-b5a2-64700ca90528.png)

**Comment by gengshan-y** [comment]
Hi, the extracted surface looks suspicious in both setups. Can you confirm the latest commit is used? If so, the recent eikonal loss update may be the culprit.

There are two ways to fix it:

(1) Without re-train the model. Remove `--mc_threshold 0` in [`scripts/render_mgpu.sh`](https://github.com/facebookresearch/banmo/blob/c4498030aeca9b056e700892788025a80953cb5a/scripts/render_mgpu.sh#L18), and run that script again.

(2) Re-train the model without eikonal loss. The original paper did not use eikonal loss. It seems eikonal loss moved the 0 isosurface inside the actual one. To reproduce the results in the paper, you might want to roll back to the commit before [this one](https://github.com/facebookresearch/banmo/commit/1d022945be987a601a36a25c27e71a8c06008bef).

**Comment by minsu1206** [comment]
Hi
sorry for late reply

(1) makes better visualization results, which are plausible and simlar to paper's figure 4. 
After re-rendering without --mc-threhsold, I also ran evaluation script again and got better ave chamfer distance (ave 8.2), which is even lower than 9.2 (indicated in Paper Table1.)

I confirmed that I just cloned your latest commit version and trained models without eikonal loss because default eikonal loss weight is 0.

Then, I have some questions.

1. Is there any additive loss function or techique which is not used at paper version's implementation? I wonder why I got better score than camera-ready version paper. Is there an possibility that some randomness like ray sampling or data fetching affect result? 

2. I want to know exact `--mc-threshold` when you rendered results for Table 1. It seems that chamfer distance is quite dependent on this hyperparameter.

I would like to hear your opinion concerning those.
Best regards.

**Comment by gengshan-y** [comment]
The experiments in the paper are before [this commit](https://github.com/facebookresearch/banmo/commit/0d507fe133443dca9bf078236a1f29748fbd73d4), so I think both fixing the bugs in the lbs implementation and replacing feature matching loss with feature rendering loss might have helped. 

The `--mc_threshold` is the default value `-0.002` for experiments in the paper.

**Comment by minsu1206** [comment]
My questions have been resolved.
Thank you for answers and sharing nice work.
